### PR TITLE
fix(table): Add border color relative to theme for accessibility

### LINKED ISSUE #25685: Contrast issue: Border between two table-{success,danger,…} rows is gray
Currently, when several rows of the same `table-<context>` class follow each other, they are separated by a gray border which looks out of place and in some cases also have contrast issues:

screen shot 2018-02-22 at 4 21 23 pm

(Note: in this screenshot the leftmost column has `rowspan="3"`)

I propose a rule similar to the following to all classes that affect table row backgrounds:

```css
tr.table-success ~ tr.table-success td, tr.table-success ~ tr.table-success th {
    border-top-color: hsla(133, 42%, 74%, 1);
}
```

It would look like this instead:

screen shot 2018-02-22 at 4 15 36 pm

### PR DESCRIPTION
**Pull request by bastienmoulia**
Related to #25685

Add a border color on table with theme classes to better accessibility contrast.

Not sure about the `-6` in `theme-color-level($color, -6)` for the border color, but it seems good to me.

capture d ecran 2018-03-01 a 22 58 32

Fixes #25685

### CODE REVIEW

**Comment by zwagoth** [comment]
As somebody who is currently fighting to make any sort of dark theme work properly, you should make the levels a variable or you force people to override them later on. This was fixed for alerts, but not for tables.

**Comment by bastienmoulia** [comment]
@zwagoth: Done

**Comment by zwagoth** [comment]
Appreciate it.

**Review by andresgalante — APPROVED** [review]
@bastienmoulia Good move, thanks!
This is following the same structure as alerts, I think we should merge it.

**Comment by XhmikosR** [comment]
@mdo: do you have any objections against this this?

**Comment by XhmikosR** [comment]
Rebased @mdo

File: scss/mixins/_table-row.scss [hunk 1]
```diff
@@ -9,6 +9,15 @@
     > td {
       background-color: $background;
     }
+
+    @if $border != null {
```
Comment (mdo) [inline scss/mixins/_table-row.scss]: Why wrap this around a `null` conditional?

Comment (XhmikosR) [reply scss/mixins/_table-row.scss]: Probably because `$border` is initialized to `null` above. And this way it's an optional parameter.

**Pull request merged**